front/src/modules/pathfinding/components/IncompatibleConstraints/IncompatibleConstrainstFilters.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, type FC, type HTMLAttributes } from 'react';
+
+import { Filter } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { toPairs } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+import type { IncompatibleConstraintTypes } from './type';
+import { getIcon } from './utils';
+
+interface IncompatibleConstraintsFiltersProps extends HTMLAttributes<unknown> {
+  data: Record<IncompatibleConstraintTypes, { count: number; enabled: boolean }>;
+  toggle: (type: string) => void;
+}
+const IncompatibleConstraintsFilters: FC<IncompatibleConstraintsFiltersProps> = (props) => {
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+  const { data, toggle, ...htmlAttrs } = props;
+  const allSelected = toPairs(data).every(([, e]) => e.enabled);
+
+  const getTitle = useCallback(
+    (label: string, enabled: boolean) => {
+      if (allSelected) {
+        return t('incompatibleConstraints.filters.focus', { name: label });
+      }
+      return enabled
+        ? t('incompatibleConstraints.filters.disable', { name: label })
+        : t('incompatibleConstraints.filters.enable', { name: label });
+    },
+    [t, allSelected]
+  );
+
+  return (
+    <div {...htmlAttrs}>
+      <Filter title={t('incompatibleConstraints.filters.title')} />
+      {toPairs(data)
+        .filter(([, type]) => type.count > 0)
```

Comment:
If you chose to pass to this component only the constraints with count > 0 (cf previous comment), you can use the props directly.

If not, I'm not sure if you think it's necessary but maybe we can extract this in a useMemo to avoid recomputing this each time the constraints list is rendered ?

```suggestion
const validFilters = useMemo(
    () => Object.entries(data).filter(([, type]) => type.count > 0),
    [data]
  );
```